source/_posts/2016-09-14-what_the_tag_series_ep1_document_object.markdown

```diff
@@ -0,0 +1,123 @@
+
+```js
+// alternative to DOMContentLoaded event
+document.onreadystatechange = function () {
+  if (document.readyState === "interactive") {
+    initApplication();
+  }
+}
+```
+
+## DOM tree accessors
+
+This accessors let you interact directly with the document, by setting the values of this properties you can modify the document and change it to your will, but remember, as uncle Ben once said:
+> With great power comes great responsibility.
+
+`document.head`
+
+This returns the head element, and it is a readonly attribute. The head element is where the metadata of the document is, for example the title, styles and scripts.
+
+`document.title`
+
+This attribute is a very simple one, it lets you get the title of the document or set it. And yes, the title is the thing is written in the browser tab.
+
+`document.body`
+
+Here is where probably most of your website's HTML is, here you can access the code directly and change it. I invite you to try to break this website, investigate about `innerHTML`, (in case you succeed just refresh the site).
+
+`document.images`
+
+Using this you can get a list of all images in the document, but it is read only, it is just an easy way to access this kind of elements, you can affect this indirectly by modifying the document and adding more images. There are more than this accessor in the list, take a look at the [definition of the images](https://html.spec.whatwg.org/multipage/dom.html#dom-document-images).
+
+`document.getElementsByName(name)`
+
+It queries the document and returns a list of elements where the attribute name is set to *name*. So if you have something like the code below and you do `document.getElementsByName('up')` you will get only the input element with *name* equal to *up*, just as simple as that!
+
+{% codeblock lang:html %}
+{% raw %}
+<form>
+  <input type="checkbox" name="up">
+  <input type="checkbox" name="down">
+</form>
+{% endraw %}
+{% endcodeblock %}
+
+
+`document.currentScript`
+
+This will just return the *script* element that is being processed by the browser. The scripts that will report only include the [classic scripts](https://html.spec.whatwg.org/multipage/webappapis.html#classic-script). Once again thanks to the friends at Mozilla for the [example](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript), in this case the script uses the *currentScript* attribute to know in which mode it is being executed, that could be pretty handy for doing some initialization.
+
+```js
+if (document.currentScript.async) {
+  console.log("Executing asynchronously");
+} else {
+  console.log("Executing synchronously");
+}
+```
+
+# What was learned today?
+
+* There is a standard that has all this things documented, I invite you to read it, I have learned a lot things myself by reading it!
```

Comment:
What do you mean by "There is a standard...". I'm not sure if you can use the word standard as a noun in english but I'm not sure about that.